sphinx_needs/layout.py

```diff
@@ -1190,6 +1231,32 @@ def _grid_content(self, colwidths, side_left, side_right, footer):
             self.node_tbody += footer_row
         node_tgroup += self.node_tbody
 
+    def _grid_customer(self, customer_format) -> None:
+        node_tgroup = nodes.tgroup(cols = len(customer_format["widths"]))
+        self.node_table += node_tgroup
+
+        for width in customer_format["widths"]:
+            node_colspec = nodes.colspec(colwidth = width)
+            node_tgroup += node_colspec
+
+        for r_k, r_v in customer_format["layout"].items():
+            row = nodes.row(classes=[r_k])
+            self.node_tbody += row
+
+            for c_k, c_v in r_v.items():
+                if "colspan" in c_v and "rowspan" in c_v:
```

Comment:
At first I had a for loop to add all elemnts, but I have to decrease these by one, so it is easier understable for the "user". Overall I have to think about, what is better:
- forward everything, without checking
- copy the know attributes